cobc/cobc.c

```diff
@@ -4983,6 +5002,20 @@ preprocess (struct filename *fn)
 		sourcename = COB_DASH;
 	} else {
 		sourcename = fn->source;
+		if( !cobc_free_format ){
+			/* If indicator is wrong on first line, switch to
+			   free format */
+			FILE* fd = fopen (sourcename, "r");
+			char buffer[7];
+			if (fd != NULL){
```

Comment:
mind the potential BOM

Reply:
if we postpone this to pplex.l (ppopen) then we can do this directly after the BOM check (only if source format is unknown).

Reply:
done